build_exercise_proj/CMakeLists.txt

```
@@ -1,24 +1,25 @@
-cmake_minimum_required(VERSION 3.1)
-project(build_exercise)
+cmake_minimum_required(VERSION 3.1) 
+project(build_exercise) 
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON) 
 
-set(_INC_FOLDER_NAME include)
-set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME})
-set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(_INC_FOLDER_NAME include) 
+set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME}) 
+set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src) 
 
 
-add_executable(
-    ${PROJECT_NAME}     
-        ${_SRC_DIR}/main.cpp
+add_executable( 
+    ${PROJECT_NAME}      
+        ${_SRC_DIR}/main.cpp 
 )
 
```

It is a good idea once the target is created (that being binary or library) to set your desired C++ standard.
Most modern compilers started defaulting to C++14, but some older ones still use C++98 (1998) as a base point.
This is bad, because anything before C++11 consired ancient. C++11 and greater is considered "modern" C++.

```suggestion
set_target_properties(
    ${PROJECT_NAME}
    PROPERTIES
        CXX_STANDARD 20
        CXX_STANDARD_REQUIRED YES
        CXX_EXTENSIONS NO
)

```

Note: this should be done for every single target (binary or library)

---

The number "20" could be changed with any standard you see fit - 11, 14, 17, 20 (and some day in the future even 23).